exercicios/para-casa/soraia/soraia.py

```diff
@@ -0,0 +1,213 @@
+        if cod_opcao == 1: incluir_nova_aluna()
+        elif cod_opcao == 2: consultar_lista_alunas()
+        elif cod_opcao == 3: consultar_faltas_aluna()
+        elif cod_opcao == 4: consultar_notas_aluna()
+        elif cod_opcao == 5: consultar_status_aprovacao()
+        elif cod_opcao == 6: print("Encerrando o programa..."); break
+        
+    
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 6).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro de 1 a 6.\n")
+            
+        return codigo_opcao
+      
+def lista_de_presenca():
+    qntdd_de_aulas = input("Quantas aulas a aluna teve? ") 
+    aulas = [] 
+    
+    
+    for contador in range(int(qntdd_de_aulas)): 
+        while True:    
+            entrada = input(f"Insira a presença da aula. Considere True=presença e False=falta, #{contador + 1}: ")
+            try: 
+                presenca = eval(entrada)
+                aulas.append(presenca) 
+                break 
+            except NameError: 
+                print("Entrada inválida. Por favor, insira True ou False.")
+                
+    return presenca
+  
+def lista_de_notas():
+    qntdd_de_notas = (input("Quantas notas quer inserir? "))
+    notas = []
+    
+    for contador in range(int(qntdd_de_notas)): 
+        
+        while True:  
+            entrada = input(f"Insira a nota {contador + 1}: ")
+            try: 
+                nota = float(entrada) 
+                notas.append(nota)
+                break
+            except:
+                print("Entrada inválida. Insira uma nota de 0 a 10. ")
+                
+    return notas
+    
+def incluir_nova_aluna():
+    print("Insira  os seguintes dados: ")
+    nome = input("Nome da aluna: ")
+    sobrenome = input("Sobrenome: ")
+    turma = input("Turma: ")
+    notas = lista_de_notas()
+    lista_presenca = lista_de_presenca()
+    
+    while True:  
+          nota_participacao = (input("Qual foi a nota de participação da aluna? "))
+          try: 
+                nota_participacao = float(nota_participacao) 
+                break
+          except:
+                print("Entrada inválida. Insira uma nota de 0 a 10. ")
+    
+    
+  
+    
```

Comment:
Acredito que aqui faltou a implementação de salvar os dados da aluna no dicionário. A parte de cima está toda correta, com as validações e captura de dados do usuário. Mas precisa ter a parte de salvar os dados da aluna. 

Ponto de atenção: É importante testar suas funções com os dados que você acabou de inserir no dicionário também. Pois aqui temos duas validações importantes, inclusão da aluna e consulta de informações desta aluna.

**Sugestão:**
```suggestion
    chave = (nome, sobrenome)
    dataset[chave] = { 
        "Turma": turma,
        "Notas": notas,
        "Presença": lista_presença,
        "Participação": nota_participação
    }   
```